### PR TITLE
Add github actions to release CSM

### PR DESCRIPTION
**Pull request by riya-kaushal7997**
# Description
Added github actions to release CSM

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1490 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x]  Tested against forked repo: 
major/ minor release: https://github.com/riya-kaushal7997/test-csm/actions/runs/12412626224
patch release: https://github.com/riya-kaushal7997/test-csm/actions/runs/12412515373

### CODE REVIEW

**Comment by shaynafinocchiaro** [comment]
@riya-kaushal7997 in your forked repo, your patch release v1.13.1 doesn't have a changelog link

**Comment by riya-kaushal7997** [comment]
> @riya-kaushal7997 in your forked repo, your patch release v1.13.1 doesn't have a changelog link

@shaynafinocchiaro , the updated code was tested with patch release v1.13.3. Workflow: https://github.com/riya-kaushal7997/test-csm/actions/runs/12412515373
Please ignore v1.13.1 as it was not tested with the updated code.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+name: Release CSM
+# Invocable as a reusable workflow
+# Can be manually triggered
+on:  # yamllint disable-line rule:truthy
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to release (major, minor, patch) Ex: 1.0.0'
```
Comment (harishp8889) [inline .github/workflows/release.yaml:9]: Please change this to `Example` instead of `Ex`

**Review by harishp8889 — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (riya-kaushal7997) [reply .github/workflows/release.yaml]: Fixed it. Thanks!

**Comment by shaynafinocchiaro** [comment]
> > @riya-kaushal7997 in your forked repo, your patch release v1.13.1 doesn't have a changelog link
> 
> @shaynafinocchiaro , the updated code was tested with patch release v1.13.3. Workflow: https://github.com/riya-kaushal7997/test-csm/actions/runs/12412515373 Please ignore v1.13.1 as it was not tested with the updated code.

@riya-kaushal7997 I see the tagged release but I don't see the v1.13.3 release here https://github.com/riya-kaushal7997/test-csm/releases

**Comment by riya-kaushal7997** [comment]
> but I don't see the v

@shaynafinocchiaro , could you please check now.

**Review by shaynafinocchiaro — APPROVED** [review]

**Pull request merged**